src/OpenToolkit.Mathematics/Color/Color3.cs

```diff
@@ -0,0 +1,1231 @@
+        }
+
+        /// <summary>
+        /// Compares whether this Color4 structure is equal to the specified Color4.
+        /// </summary>
+        /// <param name="other">The Color4 structure to compare to.</param>
+        /// <returns>True if both Color4 structures contain the same components; false otherwise.</returns>
+        [Pure]
+        public bool Equals(Color3<T> other)
+        {
+            return X.Equals(other.X) && Y.Equals(other.Y) && Z.Equals(other.Z);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="string"/> that describes this Color4 structure.
+        /// </summary>
+        /// <returns>A <see cref="string"/> that describes this Color4 structure.</returns>
+        [Pure]
+        public override string ToString()
+        {
+            return $"({X}, {Y}, {Z})";
+        }
+
+        /// <summary>
+        /// Compares whether this Color4 structure is equal to the specified object.
+        /// </summary>
+        /// <param name="obj">An object to compare to.</param>
+        /// <returns>True obj is a Color4 structure with the same components as this Color4; false otherwise.</returns>
+        [Pure]
+        public override bool Equals(object obj)
+        {
+            return obj is Color3<T> other && Equals(other);
+        }
+
+        /// <summary>
+        /// Calculates the hash code for this Color4 structure.
+        /// </summary>
+        /// <returns>A System.Int32 containing the hashcode of this Color4 structure.</returns>
+        [Pure]
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                var hashCode = X.GetHashCode();
+                hashCode = (hashCode * 397) ^ Y.GetHashCode();
+                hashCode = (hashCode * 397) ^ Z.GetHashCode();
+                return hashCode;
+            }
+        }
+    }
+
+    /// <summary>
+    /// Factory for conversion between color types.
+    /// </summary>
+    public static class Color3
+    {
+        /// <summary>
+        /// Adds all the elements of left to all the elements of right.
+        /// </summary>
+        /// <param name="left">The left hand operand.</param>
+        /// <param name="right">The right hand operand.</param>
+        /// <returns>The element generated by adding the two colors together.</returns>
+        public static Color3<Rgb> Add(this in Color3<Rgb> left, in Color3<Rgb> right) =>
+            new Color3<Rgb>(left.X + right.X, left.Y + right.Y, left.Z + right.Z);
+
+        /// <summary>
+        /// Subtracts all the elements of right from all the elements of left.
+        /// </summary>
+        /// <param name="left">The left hand operand.</param>
+        /// <param name="right">The right hand operand.</param>
+        /// <returns>The element generated by subtracting all elements of right with all elements from left.</returns>
+        public static Color3<Rgb> Subtract(this in Color4<Rgba> left, in Color3<Rgb> right) =>
```

Comment:
Wrong parameter type for the left side here.

Reply:
Everything here will be solves in the next commit